modules/shared/src/main/scala/io/constellationnetwork/currency/dataApplication/package.scala

```diff
@@ -33,9 +35,11 @@ trait DataCalculatedState
 
 case class DataState[A <: DataOnChainState, B <: DataCalculatedState](
   onChain: A,
-  calculated: B
+  calculated: B,
+  sharedArtifacts: Option[SortedSet[SharedArtifact]] = None
```

Comment:
I think it should be just `SortedSet[SharedArtifact] = SortedSet.empty` 🤔 we don't need this optional type I believe. it adds nothing here 🤔 we just need an optional type in the currency snapshot schema, to deal with backward compatibility.

Reply:
👍🏼